x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/planner/QueryFolder.java

```diff
@@ -155,13 +163,27 @@ protected PhysicalPlan rule(ProjectExec project) {
                         if (e instanceof ScalarFunction) {
                             processors.put(attr, ((ScalarFunction) e).asPipe());
                         }
+
+                        if (e instanceof NamedExpression) {
+                            attributeId = ((NamedExpression) e).toAttribute().id();
+                        }
                     }
+
+                    final NameId finalAttributeId = attributeId;
+
+                    queryC.fields().stream().filter(f -> f.attribute().id().equals(finalAttributeId)).findFirst().ifPresent(f -> {
+                        fields.add(f);
+                        nonProjectedFields.remove(f);
+                    });
```

Comment:
This looks better though a regular for/loop without the stream boilerplate code:

```java
for (field : queryC.fields()) {
  if (f.attribute().id.equals(attributeId)) {
       fields.add(f);
       nonProjectedFields.remove(f);
  }
}
```

This can be improved by moving the equality check in `FieldInfo`, say : `FieldInfo.match(Attribute)`

Reply:
Yeah, the boilerplate around streams is pretty ugly indeed. Converting it to a loop...